### PR TITLE
Reverse gist_rainbow colors to match order of manual presets

### PR DESCRIPTION
**Pull request by rosteen**
@camipacifici noticed that the RGB colors at n>5 were reversed compared to at n<=5. This fixes the problem.

Before:

image

After:

Screenshot 2024-02-27 at 11 28 31 AM

### CODE REVIEW

File: jdaviz/configs/default/plugins/plot_options/plot_options.py [hunk 1]
```diff
@@ -808,6 +808,7 @@ def apply_RGB_presets(self):
             cmap = matplotlib.colormaps['gist_rainbow'].resampled(n_visible)
             preset_colors = [matplotlib.colors.to_hex(cmap(i), keep_alpha=True) for
                              i in range(n_visible)]
+            preset_colors.reverse()
```
Comment (pllim) [inline jdaviz/configs/default/plugins/plot_options/plot_options.py:811]: Why not reverse it in the list comprehension? Also why is reverse needed here to fix the bug?
```suggestion
                             i in range(n_visible + 1, -1, -1)]
```

Comment (kecnry) [reply jdaviz/configs/default/plugins/plot_options/plot_options.py:811]: or just `cmap(n_visible - i)` 🤷‍♂️  I don't think any of these are expensive, so whatever you think is most readable (and an inline comment explaining the reason for reversal might help future-us)

Comment (rosteen) [reply jdaviz/configs/default/plugins/plot_options/plot_options.py:811]: I pushed a change but it seems like this PR didn't pick it up? I wonder if Github PRs are on the fritz right now.

**Comment by pllim** [comment]
Test failure is related.

**Comment by rosteen** [comment]
> Test failure is related.

Yup, just caught that and fixed it.

**Comment by rosteen** [comment]
Whew, Github picked up the commits after that push.

**Review by pllim — COMMENTED** [review]
The diff LGTM but I don't grok the test. I'll defer approval to the devs who worked on the original RGB stuff. Thanks!

**Comment by rosteen** [comment]
It's testing the same thing as before I just reorganized it slightly to not have to think about skipping an index in reverse order.

**Review by kecnry — APPROVED** [review]
Makes sense to me - thanks!

Just wanted to link to https://github.com/spacetelescope/jdaviz/pull/2568 in case we need a paper trail - it was that change that necessitated the flip here.

**Pull request merged**